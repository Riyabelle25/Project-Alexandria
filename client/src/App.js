import "./styles/App.css";
import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Button, CircularProgress } from "@material-ui/core";
import Video from "./views/roomCall";
import Error from "./views/Error";
import Sidebar from "./components/Sidebar";
import Chat from "./components/Chat";
import Dms from "./components/Dms";
import Login from "./views/Login";
import Home from "./views/Home";
import db, { auth } from "./app/firebase";
import ENDPOINT from "./app/env";
import { login, logout, selectUser } from "./slice/userSlice";
import Modal from "react-bootstrap/Modal";
import socketIOClient from "socket.io-client";
// const ENDPOINT = "https://alexandria-server.azurewebsites.net";

function App() {
  const dispatch = useDispatch();
  const user = useSelector(selectUser);

  useEffect(() => {
    auth.onAuthStateChanged((authUser) => {
      if (authUser) {
        dispatch(
          login({
            uid: authUser.uid,
            photo: authUser.photoURL,
            email: authUser.email,
            displayName: authUser.displayName,
          })
        );

        if (auth.currentUser.email === "riyabelle25@gmail.com") {
          db.collection("users")
            .doc(auth.currentUser.uid)
            .set({ userName: "bot" });
        } else {
          db.collection("users")
            .doc(auth.currentUser.uid)
            .set({ userName: auth.currentUser.displayName });
        }
      } else {
        dispatch(logout());
      }
    });
  }, [dispatch]);

  const [response, setResponse] = useState("");
  const [showModal, setShowModal] = useState(false);

  const socket = socketIOClient(ENDPOINT, { secure: true });
  socket.on("want-to-meet", (data) => {
    console.log("want-to-meet says server", data.receiver.userID);
    if (data.receiver.userID === auth.currentUser.uid) {
      console.log(data);
      setResponse(data);
      setShowModal(true);
    }
  });

  useEffect(() => {
    console.log(response);
    console.log(showModal);
  }, [response, showModal]);

  const handleClose = () => setShowModal(false);
  //   console.log(`${response.sender}, ${response.receiver}`) {sender:data.sender,receiver:data.receiver}

  return (
    <div className="app">
      {showModal ? (
        <Modal
          show={showModal}
          onHide={handleClose}
          style={{ zIndex: "999999" }}
        >
          <Modal.Header closeButton>
            <Modal.Title>You have a notification!</Modal.Title>
          </Modal.Header>
          <Modal.Body style={{ height: "200px", textAlign: "center" }}>
            <p>
              Hi there, {response.sender.displayName} wants to meet with you!
            </p>
          </Modal.Body>
          <Modal.Footer className="div-send-msg">
            <Button
              variant="contained"
              color="primary"
              style={{ marginRight: "3%" }}
              onClick={() => (window.location.href = response.meet)}
            >
              Join!
            </Button>
            <Button variant="contained" color="primary" onClick={handleClose}>
              Dismiss call
            </Button>
          </Modal.Footer>
        </Modal>
      ) : null}
      <Router>
        <Switch>
          <Route path="/" exact component={Login}>
            {console.log(user)}
            {user != null ? (
              user ? (
                <Home />
              ) : (
                <Login />
              )
            ) : (
              <div className="loader ">
                {" "}
                <CircularProgress color="secondary" size={100} />{" "}
              </div>
            )}
          </Route>
          <Route path="/room/:name">
            {user != null ? (
              user ? (
                <>
                  <Sidebar />
                  <Chat />
                </>
              ) : (
                <Login />
              )
            ) : (
              <div className="loader">
                {" "}
                <CircularProgress color="secondary" size={100} />{" "}
              </div>
            )}
          </Route>

          <Route
            exact
            path="/meeting/room/:name"
            render={(props) => <Video {...props} />}
          />
          <Route
            exact
            path="/meeting/chat/:name"
            render={(props) => <Video {...props} />}
          />
          <Route path="/chat/:name">
            {user != null ? (
              user ? (
                <Dms />
              ) : (
                <Login />
              )
            ) : (
              <div className="loader ">
                {" "}
                <CircularProgress color="secondary" size={100} />{" "}
              </div>
            )}
          </Route>
          <Route exact path="/:random">
            {" "}
            <Error />{" "}
          </Route>
        </Switch>
      </Router>
    </div>
  );
}
export default App;
