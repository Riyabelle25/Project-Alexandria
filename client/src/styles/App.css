.app {
  display: flex;
  background-color: rgb(13, 1, 27);
  min-height: 100vh;
}
.App{
  text-align: center;
  background-color: rgb(13, 1, 27);  
}
a, a:visited, a:active, a:hover, a:link{
color: white;
}

.App-logo {
  height: 60vmin;
  width: 40vmin;
  pointer-events: none;
}

.loader{
  height: 100vh; /* Magic here */
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;

}

.App-header {
  background-color: rgb(13, 1, 27);
  min-height: 100vh;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
} 

 .App-link {
  color: #61dafb;
}

