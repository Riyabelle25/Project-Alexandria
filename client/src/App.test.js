import App from './App';
import React from "react";
import { shallow, mount } from "enzyme";
// import Account from "./Account";
// import toJson from "enzyme-to-json";

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// it("renders correctly", () => {
//   const wrapper = mount(<App />);
//   expect(wrapper.state("error")).toEqual(null);
// });


 it("renders without crashing", () => {
  shallow(<App />);
});