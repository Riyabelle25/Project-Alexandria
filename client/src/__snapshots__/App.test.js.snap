// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `<App />`;
